src/arrayMethodSort.js

```diff
@@ -1,11 +1,37 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (callback) {
+    if (callback) {
+      for (let i = 0; i < this.length; i++) {
+        for (let j = 0; j < this.length - 1; j++) {
+          const result = callback(this[j], this[j + 1]);
+
+          if (result > 0) {
+            const temp = this[j];
+
+            this[j] = this[j + 1];
+            this[j + 1] = temp;
+          }
+        }
+      }
+    } else if (callback === undefined) {
```

Comment:
here you duplicate part of the logic if the callback is undefined

instead, set default value for callback argument
```
function(callback = defaultCallback)
```
define default callback above the current function